### PR TITLE
Feature: select update

### PR DESCRIPTION
**Pull request by SteliosKon**
## Description

Update our old select component utilizing the react-select library.

New additions:

- Multi-value selection
- Clear selection
- Inner search functionality
- Options list is now properly styled

## Changes

This PR contains breaking changes, that need to be followed by client changes.

### CODE REVIEW

File: src/components/FormElements/Select/components/CustomValueContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  return (
+    <ValueContainer {...props} selectProps={selectProps}>
+      {Children.map(children, (child) => {
+        return child ? (
+          child
+        ) : props.hasValue ? (
+          <SingleValue
+            {...commonProps}
+            isDisabled={selectProps.isDisabled}
```
Comment (ThrasyvoulosKafasis) [inline src/components/FormElements/Select/components/CustomValueContainer.tsx:34]: isDisabled, getClassNames and innerProps are the same between SingleValue and Placeholder component. Can't be added in the common props?

File: src/components/FormElements/Select/styles.ts [hunk 2]
```diff
@@ -1,47 +1,46 @@
 import { css, SerializedStyles, Theme } from "@emotion/react";
-import { inputBaseStyles, inputContainerBaseStyles } from "../styles";
 import { InputSize } from "../Input/Input";
 
+const inputHeight = {
```
Comment (ThrasyvoulosKafasis) [inline src/components/FormElements/Select/styles.ts:4]: Import inputHeight from src/components/FormElements/styles.ts

File: src/components/FormElements/Select/styles.ts [hunk 3]
```diff
@@ -1,47 +1,46 @@
...
+  sm: "1.875rem",
+  md: "2.5rem",
+  lg: "3rem",
+};
+
+const optionPadding = {
```
Comment (ThrasyvoulosKafasis) [inline src/components/FormElements/Select/styles.ts:10]: Use rem

**Review by ThrasyvoulosKafasis — CHANGES_REQUESTED** [review]

File: src/components/FormElements/Select/styles.ts (hunk 3 above)
Comment (SteliosKon) [reply src/components/FormElements/Select/styles.ts:5]: Fixed [here](https://github.com/epignosis/gnosis/pull/194/commits/63f291f365316d8796425f9f47f7c10b228ecd02)

File: src/components/FormElements/Select/styles.ts (hunk 2 above)
Comment (SteliosKon) [reply src/components/FormElements/Select/styles.ts]: Fixed [here](https://github.com/epignosis/gnosis/pull/194/commits/63f291f365316d8796425f9f47f7c10b228ecd02)

File: src/components/FormElements/Select/components/CustomValueContainer.tsx (hunk 1 above)
Comment (SteliosKon) [reply src/components/FormElements/Select/components/CustomValueContainer.tsx]: No, we cannot add them to common props. I simplified them [here](https://github.com/epignosis/gnosis/pull/194/commits/a0d11d33112d157d624c7f818f6d09512caed70b)

**Review by ThrasyvoulosKafasis — APPROVED** [review]

**Pull request merged**